src/test/kotlin/fi/metatavu/timebank/api/test/functional/tests/VacationRequestsTest.kt

```diff
@@ -0,0 +1,104 @@
+    fun testVacationRequest() {
+        createTestBuilder().use { testBuilder ->
+            testBuilder.userA.vacationRequests.createVacationRequests(
+                vacationRequest = VacationRequest(
+                    person = 1,
+                    startDate = LocalDate.now().toString(),
+                    endDate = LocalDate.now().plusDays(1).toString(),
+                    days = 2,
+                    type = VacationType.VACATION,
+                    message = "Lomaa!!!",
+                    projectManagerStatus = VacationRequestStatus.PENDING,
+                    hrManagerStatus = VacationRequestStatus.PENDING,
+                    createdAt = getODT(getThirtyDaysAgoThirdWeek()[1].atStartOfDay()),
+                    updatedAt = getODT(getThirtyDaysAgoThirdWeek()[1].atStartOfDay()),
+                )
+            )
+            var vacations = testBuilder.manager.vacationRequests.listVacationRequests(
+                personId = 1,
+                after = LocalDate.now().toString()
+            )
+
+            val id = vacations[0].id!!
+
+            assertEquals(1, vacations.size)
+            assertTrue(vacations.find { it.id == id } != null)
+            assertTrue(vacations.find { it.projectManagerStatus == VacationRequestStatus.PENDING} != null)
+            assertTrue(vacations.find { it.lastUpdatedBy == UUID.fromString("7276979e-2f08-4d52-9541-0d10aa3806fe") } != null)
+
+            testBuilder.manager.vacationRequests.updateVacationRequests(
+                id = id,
+                vacationRequest = VacationRequest(
+                    person = 1,
+                    startDate = LocalDate.now().toString(),
+                    endDate = LocalDate.now().plusDays(1).toString(),
+                    days = 2,
+                    type = VacationType.VACATION,
+                    message = "Lomaa!!!",
+                    projectManagerStatus = VacationRequestStatus.APPROVED,
+                    hrManagerStatus = VacationRequestStatus.PENDING,
+                    createdAt = getODT(getThirtyDaysAgoThirdWeek()[1].atStartOfDay()),
+                    updatedAt = getODT(getThirtyDaysAgoThirdWeek()[2].atStartOfDay()),
+                )
+            )
+            vacations = testBuilder.manager.vacationRequests.listVacationRequests(
+                personId = 1,
+                after = LocalDate.now().toString()
+            )
+
+            assertEquals(1, vacations.size)
+            assertTrue(vacations.find { it.id == id } != null)
+            assertTrue(vacations.find { it.projectManagerStatus == VacationRequestStatus.APPROVED } != null)
+            assertTrue(vacations.find { it.hrManagerStatus == VacationRequestStatus.PENDING } != null)
+            assertTrue(vacations.find { it.lastUpdatedBy == UUID.fromString("50bd84bc-d7f7-445f-b98c-4f6a5d27fb55") } != null)
+
+            vacations.forEach { vacation ->
+                testBuilder.manager.vacationRequests.clean(vacation)
+            }
```

Comment:
JAX-RS Functional TestBuilder should handle the cleaning by its own so this should be redundant.
By adding all of the created resources by each TestBuilderResource into a list of `AutoCloseables` TestBuilder will call each resources `::clean` method in a reverse order of when they've been added to the list, e.g. no foreign key constraint violations should happen. 
Reference of that can be found for example in: https://github.com/Metatavu/oss-surveys-api/blob/develop/src/test/kotlin/fi/metatavu/oss/api/test/functional/impl/SurveysTestBuilderResource.kt#L22-L34

This same thing should be applied to all of the tests in this project.